src/Database/LMDB/Simple/Codec.hs

```diff
@@ -0,0 +1,107 @@
+{-# LANGUAGE Rank2Types #-}
+
+module Database.LMDB.Simple.Codec (
+    -- * Types
+    Codec (..)
+    -- * Serialisation and deserialisation with explicit dictionaries
+  , deserialise
+  , deserialiseIncremental
+  , serialise
+  , serialiseIncremental
+    -- * Marshalling and others
+  , marshalIn
+  , marshalOut
+  , marshalOutBS
+  , serialiseLBS
+  , serialiseBS
+  , peekVal
+  , pokeVal
+  ) where
+
+import           Control.Exception             (throw)
+import           Control.Monad                 ((>=>))
+import           Control.Monad.ST
+
+import qualified Data.ByteString               as BS
+import qualified Data.ByteString.Builder       as BS
+import qualified Data.ByteString.Lazy          as LBS
+import qualified Data.ByteString.Lazy.Internal as LBS
+import           Data.ByteString.Unsafe        (unsafeUseAsCStringLen)
+
+import           Foreign                       (Ptr, castPtr, peek, poke)
+
+import           Codec.CBOR.Decoding
+import           Codec.CBOR.Encoding
+import qualified Codec.CBOR.Read               as CBOR.Read
+import qualified Codec.CBOR.Write              as CBOR.Write
+
+import           Database.LMDB.Raw
+
+{-------------------------------------------------------------------------------
+  Types
+-------------------------------------------------------------------------------}
+
+-- | A @'Codec' a@ contains a CBOR encoder and a CBOR decoder for values of type
```

Comment:
I like to use it for providing links in the `haddock`-generated documentation. Technically, it's not really necessary because the comment is on the `Codec` datatype itself, but I do not mind having the link there